### PR TITLE
Feature: save records by page in localstorage

### LINKED ISSUE #369: DiagramTable: Make "Records per page" persistent
**Describe the bug**

In the `ModelsPage`: when I change the "Records per page", then open a diagram, then go back to the `ModelsPage`: the "Records per page" is reset to its default value.

I expect the "Records per page" to be persisted (at least in memory), so that it gets restored when I return to the `ModelsPage`.

**To Reproduce**

https://github.com/ditrit/leto-modelizer/assets/20184007/a1aeb60f-5a04-45db-99c2-258200bf5f10

**Application**

- `leto-modelizer` commit hash: `f59e63e9d861fb5d0f0849dc7983be3a76bd779a`.

### PR DESCRIPTION
**Pull request by Zorin95670**
Resolved #369.

### CODE REVIEW

File: src/models/UserSettings.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Class to store all user settings.
+ */
+class UserSettings {
+  /**
+   * Default constructor.
+   * @param {object} [props] - Object that contains all properties to set.
+   * @param {string} [props.displayType] - Display type for diagrams, only 'grid' or 'table' are
+   * accepted.
```
Comment (k-y0u) [inline src/models/UserSettings.js:9]: Start a new sentence from "only" and break line before.

File: src/models/UserSettings.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+   * @param {number} [props.recordsByPage] - Records by page for diagrams table.
```
Comment (k-y0u) [inline src/models/UserSettings.js:10]: Number of records...

File: src/composables/Settings.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import UserSettings from 'src/models/UserSettings';
+
+/**
+ * Get user settings by key.
```
Comment (k-y0u) [inline src/composables/Settings.js:4]: user's setting
(owner + singular)

File: src/composables/Settings.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * @param {string} key - User setting key.
+ * @returns {string|number|boolean} User settings.
```
Comment (k-y0u) [inline src/composables/Settings.js:6]: same

File: src/composables/Settings.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import UserSettings from 'src/models/UserSettings';
+
+/**
+ * Get user settings by key.
+ * @param {string} key - User setting key.
```
Comment (k-y0u) [inline src/composables/Settings.js:5]: just `setting key`

File: src/composables/Settings.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ * @returns {string|number|boolean} User settings.
+ */
+export function getUserSettings(key) {
```
Comment (k-y0u) [inline src/composables/Settings.js:8]: why "by key" and not all ?

File: src/composables/Settings.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  return new UserSettings(JSON.parse(localStorage.getItem('user-settings')))[key];
+}
+
+/**
+ * Set user settings key with provided value.
+ * @param {string} key - User setting key.
+ * @param {string|number|boolean} value - User setting value.
+ */
+export function setUserSettings(key, value) {
```
Comment (k-y0u) [inline src/composables/Settings.js:17]: same

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/models/UserSettings.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+   */
+  constructor(props = {
+    displayType: 'table',
+    recordsByPage: 5,
+  }) {
+    /**
+     * Display type for diagrams, only 'grid' or 'table' are accepted.
+     * @type {string}
+     */
+    this.displayType = props?.displayType || 'table';
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/models/UserSettings.js:20]: see together won't fix.

File: src/models/UserSettings.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Class to store all user settings.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/models/UserSettings.js:2]: done

File: src/models/UserSettings.js (hunk 1 above)
Comment (Zorin95670) [reply src/models/UserSettings.js:9]: nope

File: src/composables/Settings.js (hunk 3 above)
Comment (Zorin95670) [reply src/composables/Settings.js:4]: done

File: src/composables/Settings.js (hunk 4 above)
Comment (Zorin95670) [reply src/composables/Settings.js:6]: done

File: src/composables/Settings.js (hunk 5 above)
Comment (Zorin95670) [reply src/composables/Settings.js:5]: done

File: src/composables/Settings.js (hunk 6 above)
Comment (Zorin95670) [reply src/composables/Settings.js:8]: see together, won't fix

File: src/composables/Settings.js (hunk 7 above)
Comment (Zorin95670) [reply src/composables/Settings.js:17]: see together, won't fix

File: src/models/UserSettings.js (hunk 1 above)
Comment (k-y0u) [reply src/models/UserSettings.js:9]: ugly

**Review by k-y0u — APPROVED** [review]

**Review by aboubakar14 — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**